### PR TITLE
Utilize PHPUnit\Framework\TestCase class to broaden compatibility with other packages

### PR DESCRIPTION
**Pull request by ph7jack**
Enhanced Compatibility with Other Packages Utilizing `PHPUnit\Framework\TestCase` Class

**1️⃣ Is this something that is wanted/needed? Did you create a discussion about it first?**

Yes, enhancing compatibility is always beneficial as it promotes interoperability with a wider array of packages. 
No discussion was created.

**2️⃣ Did you create a branch for your fix/feature? (Master branch PR's will be closed)**

Yes, a separate branch named `add-testcase-compatibility` has been created for this purpose.

**3️⃣ Does it contain multiple, unrelated changes? Please separate the PRs out.**

No, this PR solely focuses on enhancing compatibility by utilizing the `PHPUnit\Framework\TestCase` class.

**4️⃣ Does it include tests? (Required)**

No, existing tests continue to pass with these changes, serving primarily to extend compatibility with other packages/projects, ensuring seamless integration as they evolve.

**5️⃣ Please include a thorough description (including small code snippets if possible) of the improvement and reasons why it's useful.**

Testing Livewire components or integrations within packages can sometimes pose challenges. This PR aims to streamline this process, enabling developers to leverage the Livewire testing infrastructure for a broader range of testing scenarios. This change is fairly straightforward yet significantly impactful.

```php
Livewire::visit(new class() extends Component
{
    public $color = '#000';

    public function render(): string
    {
        return <<<'BLADE'
            <div>
                <x-color-picker label="Color Picker" name="color" wire:model="color" />
            </div>
        BLADE;
    }
})
    ->waitForAlpineJs()
    ->assertInputValue('color', '#000');
```

By adhering to PHPUnit's native TestCase class, we foster a more standardized approach, which in turn facilitates integration with other packages. This is particularly useful for projects that rely on a variety of packages and aim for a cohesive working environment.

Thanks for this awesome package! 🙌

### CODE REVIEW

**Comment by PH7-Jack** [comment]
I believe also, it would be interesting to expose the livewire test case. https://github.com/livewire/livewire/blob/07de2790f0aaf47c53f3a8cf942139a54b4609d8/tests/BrowserTestCase.php

**Comment by calebporzio** [comment]
I'm sorry, I don't quite understand what's preventing people from using these testing utilities in their current package's test suite?

**Comment by PH7-Jack** [comment]
> I'm sorry, I don't quite understand what's preventing people from using these testing utilities in their current package's test suite?

Hi, it's because The `Livewire::visit()`, is not working with PHPUnit 10, and inside packages.
For my package [wireui](https://github.com/wireui/wireui), the usage of `Livewire::visit()` will be excellent. Tests will be easier and simpler to create.
However, the lines 99 and 123 block it. 

Line 99: only accept your test case class 
Line 123: PHPUnit 10 needs the method name parameter in the class constructor.

**Comment by lrljoe** [comment]
I'd second this being a pain presently, very keen to use some of the native testing methods for packages, but the current approach stops me in my tracks when it comes to Dusk/Livewire::visit.

Might be that I'm missing something tho!

File: src/Features/SupportTesting/DuskTestable.php [hunk 1]
```diff
@@ -120,7 +120,7 @@ static function loadTestComponents()
 
             $components = null;
 
-            try { (new $testClass)->$method(); } catch (\Exception $e) {
+            try { (new $testClass($method))->$method(); } catch (\Exception $e) {
```
Comment (crynobone) [inline src/Features/SupportTesting/DuskTestable.php:123]:
```suggestion
            try { 
                if (\Orchestra\Testbench\phpunit_version_compare('10.0', '>=') {
                    (new $testClass($method))->$method(); 
                } else {
                    (new $testClass())->$method(); 
                }
            } catch (\Exception $e) {
```

Support both PHPUnit 9 and PHPUnit 10.

**Comment by crynobone** [comment]
Looks good, the failing tests can be fixed by adding a `pause()`:

```diff
        ->assertDontSee('Toggle Me!')
        ->waitForLivewire()->click('@toggle')
+       ->pause(500)
        ->assertSee('Toggle Me!')
        ;
```

**Review by crynobone — APPROVED** [review]

**Comment by devajmeireles** [comment]
> Looks good, the failing tests can be fixed by adding a `pause()`:
> 
> ```diff
>         ->assertDontSee('Toggle Me!')
>         ->waitForLivewire()->click('@toggle')
> +       ->pause(500)
>         ->assertSee('Toggle Me!')
>         ;
> ```
> 
> I believe it depends on Livewire event cycle where there's a small wait between click and DOM updates.

I think `waitForText` is most appropriate in this case.

**Comment by crynobone** [comment]
Wait for text would be good but it doesn't add to the assertion count.

**Comment by PH7-Jack** [comment]
I believe this test should be fixed in another PR to livewire, since it's not part of this feature.

**Comment by calebporzio** [comment]
Gotcha, sounds good!

**Pull request merged**